ESP8266Interface.cpp

```diff
@@ -43,8 +43,11 @@ ESP8266Interface::ESP8266Interface(PinName tx, PinName rx, bool debug)
     memset(_cbs, 0, sizeof(_cbs));
     memset(ap_ssid, 0, sizeof(ap_ssid));
     memset(ap_pass, 0, sizeof(ap_pass));
+    ap_sec = NSAPI_SECURITY_UNKNOWN;
 
     _esp.attach(this, &ESP8266Interface::event);
+    _esp.setTimeout(ESP8266_CONNECT_TIMEOUT);
+    (void)_esp.reset();
```

Comment:
Does this mean we now send a command in the constructor for the first time?

For general debuggability, probably want to avoid that, and see my other comment about taking reset out of connect.

But if we do go down that "issue commands in constructor" path, then the version check could be here too.